### PR TITLE
Fix path transform for sibling ancestor moves

### PR DESCRIPTION
**Pull request by jason-krypton**
#### How does this change work?

My previous fix https://github.com/ianstormtaylor/slate/pull/3675 did not go far enough. Extending that change to also fix sibling moves around ancestors.

#### Have you checked that...?

- [X] The new code matches the existing patterns and styles.
- [X] The tests pass with `yarn test`.
- [X] The linter passes with `yarn lint`. (Fix errors with `yarn fix`.)
- [X] The relevant examples still work. (Run examples with `yarn start`.)

#### Does this fix any issues or need any specific reviewers?

Reviewers: @CameronAckermanSEL

### CODE REVIEW

File: packages/slate/src/interfaces/path.ts [hunk 1]
```diff
@@ -386,12 +386,14 @@ export const Path = {
             const copy = onp.slice()
 
             if (Path.endsBefore(op, onp) && op.length < onp.length) {
-              const i = Math.min(onp.length, op.length) - 1
-              copy[i] -= 1
+              copy[op.length - 1] -= 1
```
Comment (jason-krypton) [inline packages/slate/src/interfaces/path.ts:389]: Note: This is just clean up. The length check in the if already guaranteed which path is less.

**Comment by jason-krypton** [comment]
@TheSpyder The sample you provided is ancestor (parent) to ends after. Did you just want to see parent to moves after instead of grandparent?

**Comment by TheSpyder** [comment]
@jason-krypton yes. I was looking at the `ancestor-to-*` tests and wondered how it would react if the direct parent was moved instead of the root ancestor. It'll probably be fine but an extra test can't hurt :)

**Comment by jason-krypton** [comment]
@TheSpyder tests added

**Comment by TheSpyder** [comment]
Thanks!

**Comment by CameronAckermanSEL** [comment]
Thanks again for the PR @jason-krypton !

**Pull request merged**